### PR TITLE
Adds a reusable user-avatar-component

### PR DESCRIPTION
**Pull request by franciscotln**
Adds a reusable user-avatar-component. Fix in the tslint, new barrels with index.ts files for easier imports. new JiraUserService, new helpers handle-for to be used after ajax requests, new helper toPayload, new helper combineMicroReducers, fixed some linter problems

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,37 +1,73 @@
...
+    .subscribe((hasAccount) => {
+      this.hasAccount = hasAccount;
+    });
   }
+
+  private getAccountAndSetJiraReqOpts(): void {
+    this.store.select('account', 'account', 'model')
+      .filter(complement(isNil))
+      .map(getHeaderOptionsByAccount)
+      .do((account: IJiraRequestOptions) => this.jiraRequestOptionsService.setOptions(account))
```
Comment (dakolech) [inline src/app/app.component.ts:57]: should be done in the `subscribe`

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,37 +1,73 @@
...
   public logout(): void {
-    this._store.dispatch(new accountActions.Logout())
+    this.store.dispatch(new accountActions.Logout());
+  }
+
+  private checkIfHasAccount(): void {
+    this.store.select('account', 'account', 'model')
+    .map((account: any) => !!account)
+    .takeUntil(this.subscriptionKiller$)
+    .subscribe((hasAccount) => {
```
Comment (dakolech) [inline src/app/app.component.ts:48]: why subscribe? it can be done with `async` pipe, which is better

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,37 +1,73 @@
...
+    this.getAndAssignUser();
   }
 
-  public ngOnInit(): void {
-    this._store.select('account', 'account', 'model')
-      .filter(Boolean)
-      .map(getHeaderOptionsByAccount)
-      .do((account: IJiraRequestOptions) => this._jiraRequestOptionsService.setOptions(account))
-      .subscribe()
+  public ngOnDestroy(): void {
```
Comment (dakolech) [inline src/app/app.component.ts:36]: Angular automatically unsubscribes all streams

File: src/app/app.component.ts [hunk 4]
```diff
@@ -1,37 +1,73 @@
...
 })
-export class AppComponent implements OnInit {
+export class AppComponent implements OnInit, OnDestroy {
+  public hasAccount: boolean;
+  public currentUser: IUser;
+
+  private subscriptionKiller$: Subject<null> = new Subject();
+
   constructor(
-    private _store: Store<any>,
```
Comment (dakolech) [inline src/app/app.component.ts:17]: you shouldn't change @miroslaw-bogacz convention

**Review by dakolech — CHANGES_REQUESTED** [review]

File: src/app/app.component.ts (hunk 2 above)
Comment (franciscotln) [reply src/app/app.component.ts:48]: Did you compare with the previous code? ;-)
Observables with async pipe cannot be a getter

File: src/app/app.component.ts (hunk 1 above)
Comment (franciscotln) [reply src/app/app.component.ts:57]: That's with @miroslaw-bogacz , I deleted the `_`

File: src/app/app.component.ts (hunk 2 above)
Comment (dakolech) [reply src/app/app.component.ts:48]: but can be an attribute

**Pull request closed without merging**